modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/PartitionSnapshotStorage.java

```diff
@@ -69,6 +72,8 @@ public class PartitionSnapshotStorage implements SnapshotStorage {
     /** Incoming snapshots executor. */
     private final Executor incomingSnapshotsExecutor;
 
+    private final long waitForMetadataCatchupMs;
```

Comment:
Additional asynchrony in `IncomingSnapshotCopier` broke a test, and it turned out that it was wrong. After a proper fix, the test started taking 3 seconds to pass because of these default 3 seconds of wait. I added ability to configure it to set it to 0 in tests. And `IncomingSnapshotCopier` is instantiated by `PartitionSnapshotStorage`, so I had to add the 'configuration' here.